File: binary-search/binary-search.py
def binarySearch(list, value):
  low = 0
  high = len(list) - 1
  
  while low <= high:
    mid = (low + high) / 2
    guess = list[mid]
    if guess == value:
      return mid
    if guess > value:
      high = mid - 1
    else:
      low = mid + 1
  return None

my_list = [0, 1, 2, 4, 6, 8, 9, 11]

print binarySearch(my_list, 4)
print binarySearch(my_list, 5)

File: binary-search/binary-search.js
const binarySearch = function(arr, value) {
  let low = 0;
  let high = arr.length - 1;
  
  while (low <= high) {
    let mid = Math.floor((high + low) / 2);

    if (value === arr[mid]) {
      return mid;
    } else if (value < arr[mid]) {
      high = mid - 1;
    } else {
      low = mid + 1;
    };
  };
};

let myList = [0, 1, 2, 4, 6, 8, 9, 11];

console.log(binarySearch(myList, 4));
console.log(binarySearch(myList, 7));
